### ISSUE TITLE
replaceError retry bug with async stop

### ISSUE THREAD

**Issue by milankinen** [staltz/xstream#24]
It seems that `.replaceError` method does not stop the errored stream before replacing it with the new one. I'm not sure whether this is a bug or expected behaviour, but if I remember our conversation with @staltz correctly, `.replaceError` should be able to "retry" the errored stream `x` by returning it from the replace function `retried = x.replaceError(() => x)`. 

Based on that I'd expect the following example to print
```
next hi
complete
```
Instead I get nothing. Is this a bug or am I just doing something wrong?

```js
import xs from "xstream"

let i = 0;

const response = xs.create({
  start: l => setTimeout(() => { 
    if (i++ === 0) {
      l.error(new Error("oops"))
    } else {
      l.next("hi")
      l.complete()
    } 
  }, 10),
  stop: () => {}
})

const retried = response.replaceError(() => response)

retried.addListener({
  next: x => console.log("next", x),
  complete: () => console.log("complete"),
  error: e => console.error("error", e)
})
```

**Comment by staltz** [comment]
I had to tweak your example code a bit, and there is indeed a bug. 
```js
var xs = xstream.default;

let i = 0;

const response = xs.create({
  id: null,
  start: function (l) {
    this.id = setTimeout(() => { 
      console.log('yep', i);
      if (i++ === 0) {
        l.error(new Error("oops"))
      } else {
        l.next("hi")
        l.complete()
      } 
    }, 10);
  },
  stop: function () { 
    if (this.id !== null) clearTimeout(this.id); 
  }
})

const retried = response.replaceError(() => response)

retried.addListener({
  next: x => console.log("next", x),
  complete: () => console.log("complete"),
  error: e => console.error("error", e)
})
```

Thanks for reporting. retrying is working in some case (just check the tests), but this one is a corner case.

**Comment by xtianjohns** [comment]
This one is interesting. Most importantly, **this bug was resolved in `v6.0.0`.**

My biggest reason for thinking this is that I can resolve @milankinen's original [issue](https://github.com/staltz/xstream/issues/24#issue-151834876) by simply providing `replaceError` with a function that will return a different stream than the stream upon which the operator is called. Check out [this bin](http://www.webpackbin.com/NkE-vuz0-) for what I think is proof of this.

There may be some sort of behavior that is unique to `replaceError` that is worth testing in 9e94ea077b29836c39d59af841e0454a356a4d72. However, I think there's good reason to believe this issue is related to #90, #91 and #104, and that for that reason it is resolved.

**Comment by staltz** [comment]
@xtianjohns did you see that if you use xstream v6.5.0 in that webpackbin, both `neverComplete` and `works` complete? So yes, I think this issue is resolved.
